crates/rethnet_evm/benches/state_debug.rs

```diff
@@ -0,0 +1,330 @@
+                        Address::from_low_u64_ne(number_of_accounts + 1).as_bytes(),
+                    )));
+                })),
+                &|| Ok(AccountInfo::default()),
+            );
+            debug_assert!(result.is_ok());
+        },
+    );
+}
+
+fn bench_modify_account_exists_without_code_no_code_change(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::modify_account(), account already exists, without code, no code change",
+        |state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            state
+                .modify_account(
+                    address,
+                    AccountModifierFn::new(Box::new(|_balance, _nonce, code| {
+                        code.take();
+                    })),
+                    &|| Ok(AccountInfo::default()),
+                )
+                .unwrap();
+        },
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            let result = state.modify_account(
+                address,
+                AccountModifierFn::new(Box::new(|_balance, &mut _nonce, _code| {})),
+                &|| Ok(AccountInfo::default()),
+            );
+            debug_assert!(result.is_ok());
+        },
+    );
+}
+
+fn bench_remove_account_with_code(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::remove_account() existing account with code",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            // TODO: figure out why the following assert is failing
+            //debug_assert!(state.basic(address).unwrap().unwrap().code.is_some());
+            let result = state.remove_account(address);
+            debug_assert!(result.is_ok());
+            debug_assert!(result.unwrap().is_some());
+        },
+    );
+}
+
+fn bench_remove_account_without_code(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::remove_account() existing account without code",
+        |state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            state
+                .modify_account(
+                    address,
+                    AccountModifierFn::new(Box::new(|_balance, _nonce, code| {
+                        code.take();
+                    })),
+                    &|| Ok(AccountInfo::default()),
+                )
+                .unwrap();
+        },
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().unwrap().code.is_none());
```

Comment:
This will always be true, as we split code and account storage

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3910/commits/64451147b184637dac362d70d374047bd0e0a822